src/zone/napa-zone.cpp

```diff
@@ -106,20 +115,30 @@ const std::string& NapaZone::GetId() const {
     return _settings.id;
 }
 
-void NapaZone::Broadcast(const std::string& source, BroadcastCallback callback) {
+void NapaZone::Broadcast(const FunctionSpec& spec, BroadcastCallback callback) {
     // Makes sure the callback is only called once, after all workers finished running the broadcast task.
     auto counter = std::make_shared<std::atomic<uint32_t>>(_settings.workers);
-    auto callOnce = [this, source, callback = std::move(callback), counter](napa_result_code code) {
+    auto callOnce = [this, callback = std::move(callback), counter](Result result) {
         if (--(*counter) == 0) {
-            NAPA_DEBUG("Zone", "Finishing broadcast script \"%s\" to zone \"%s\"", source.c_str(), _settings.id.c_str());
```

Comment:
why do you remove this line of NAPA_DEBUG?

Reply:
We are no longer holding the `source` in the callback. As for `spec`, it contains a unique pointer so we can not capture it here.

If we get no info about the current context, it does not help to output a log just saying "Finishing broadcast."